datacube/index/postgis/_datasets.py

```diff
@@ -747,25 +767,27 @@ def _do_search_by_product(self, query, return_fields=False, select_field_names=N
                                           if not field.affects_row_selection)
                 else:
                     select_fields = tuple(dataset_fields[field_name]
-                                          for field_name in select_field_names)
+                                          for field_name in select_field_names
+                                          if field_name in dataset_fields)
```

Comment:
Should there be some sort of messaging if a field name is no in `dataset_fields`? I can't imagine that would be intentional.

Reply:
As noted above, it can be intentional.